### PR TITLE
Add raw socket transport

### PR DESCRIPTION
**Pull request by danmarcab**
Implementation of rawsocket transport as defined in WAMP advanced profile:
https://github.com/tavendo/WAMP/blob/master/spec/advanced.md#rawsocket-transport

I tried to document every binary frame to make it understandable.
Let me know what you guys think.

There are some bits missing:
* The peer sets a max_length for messages, if a message longer than that is received it should close the connection.
* The router also sets a max_length for messages, before sending a message we should check for that limit and discard the message (and probably inform the caller).
* When the handshake is not accepted the router returns an error code, we have to translate it into meaningful errors.

### CODE REVIEW

**Comment by mtanzi** [comment]
Overall it looks good to me

**Comment by yelps** [comment]
Great! Well done!

File: lib/crossbar.ex [hunk 1]
```diff
@@ -215,8 +234,8 @@ defmodule Crossbar do
     {:ok, state}
   end
 
-  def handle_info({Spell.Transport.WebSocket, _pid,
-                   {:terminating, {:remote, :closed}}}, _state) do
+  def handle_info({transport, _pid, {:terminating, {:remote, :closed}}}, _state)
+                  when transport in [Spell.Transport.WebSocket, Spell.Transport.RawSocket] do
```
Comment (yelps) [inline lib/crossbar.ex]: I think the guard clause is not necessary in this case.

**Comment by mtanzi** [comment]
:+1:

**Pull request merged**